From 830e614b3a1087f3d904be550b06884a946f7704 Mon Sep 17 00:00:00 2001
From: Drew Moseley <drew_moseley@mentor.com>
Date: Sat, 6 Dec 2014 11:32:31 -0500
Subject: [PATCH 7/7] init-install.sh: Add extra sync time.

On my Steppe Eagle LC board with an eMMC disk, the
legacy mode install occasionally fails with an invalid
partition table.  These extra sync calls seem to work
around the failure.  Note this is only a workaround,
the real fix is likely kernel-side.

Signed-off-by: Drew Moseley <drew_moseley@mentor.com>
---
 meta/recipes-core/initrdscripts/files/init-install.sh | 10 ++++++++++
 1 file changed, 10 insertions(+)

Upstream-Status: Inappropriate [configuration]

diff --git a/meta/recipes-core/initrdscripts/files/init-install.sh b/meta/recipes-core/initrdscripts/files/init-install.sh
index 842707d..f227c3e 100644
--- init-install.sh
+++ init-install.sh
@@ -142,26 +142,34 @@ echo "Swap partition size:   $swap_size MB ($swap)"
 echo "*****************"
 echo "Deleting partition table on /dev/${device} ..."
 dd if=/dev/zero of=/dev/${device} bs=512 count=2
+sync; udevadm settle
 
 echo "Creating new partition table on /dev/${device} ..."
 parted /dev/${device} mklabel msdos
+sync; udevadm settle
 
 echo "Creating boot partition on $bootfs"
 parted /dev/${device} mkpart primary 0% $boot_size
+sync; udevadm settle
 
 echo "Creating rootfs partition on $rootfs"
 parted /dev/${device} mkpart primary $rootfs_start $rootfs_end
+sync; udevadm settle
 
 echo "Creating swap partition on $swap"
 parted /dev/${device} mkpart primary $swap_start 100%
+sync; udevadm settle
 
 parted /dev/${device} print
+sync; udevadm settle
 
 echo "Formatting $bootfs to ext3..."
 mkfs.ext3 $bootfs
+sync; udevadm settle
 
 echo "Formatting $rootfs to ext3..."
 mkfs.ext3 $rootfs
+sync; udevadm settle
 
 echo "Formatting swap partition...($swap)"
 mkswap $swap
@@ -177,6 +185,7 @@ mkdir /src_root
 mkdir -p /boot
 
 # Handling of the target root partition
+sync; udevadm settle
 mount $rootfs /tgt_root
 mount -o rw,loop,noatime,nodiratime /run/media/$1/$2 /src_root
 echo "Copying rootfs files..."
@@ -193,6 +202,7 @@ umount /tgt_root
 umount /src_root
 
 # Handling of the target boot partition
+sync; udevadm settle
 mount $bootfs /boot
 echo "Preparing boot partition..."
 if [ -f /etc/grub.d/00_header ] ; then
-- 
1.9.1

